### PR TITLE
enhance: scann support iterator

### PR DESCRIPTION
**Pull request by cqy123456**
issue:https://github.com/zilliztech/knowhere/issues/909

### CODE REVIEW

File: thirdparty/faiss/faiss/IndexScaNN.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #include <faiss/utils/Heap.h>
 #include <faiss/utils/distances.h>
 #include <faiss/utils/utils.h>
+#include <faiss/FaissHook.h>
```
Comment (cqy123456) [inline thirdparty/faiss/faiss/IndexScaNN.cpp:10]: remove #include <faiss/FaissHook.h>

**Comment by alexanderguzhva** [comment]
lgtm overall

**Review by foxspy — APPROVED** [review]
/lgtm

**Pull request merged**